packages/neos-ts-interfaces/src/index.ts

```diff
@@ -0,0 +1,141 @@
+export type NodeContextPath = string;
+export type NodeTypeName = string;
+
+// TODO: for some reason (probably due to immer) I can not use ReadonlyArray here
```

Comment:
Maybe use `type` here:
```
type NodeChild = {...};
type NodeChildren = ReadonlyArray<NodeChild>;
```
So if you iterate over `NodeChildren` an element has the type `NodeChild` instead of `{....}`. It's a little bit more expressive.

I don't think it will fix the immer issue, though. Let's have a look at this later :)

Reply:
Will give it a try. Perhaps we could even have a pair coding hangout to finalize this PR. I won't have the time today though.

Reply:
tslint tells me `Use an interface instead of a type literal. [interface-over-type-literal]`

Reply:
We could do 
```
interface NodeChild {...}
type NodeChildren = ReadonlyArray<NodeChild>;
```
but it does not feel consistent somehow..